roles/docker/tasks/set_facts_dns.yml

```diff
@@ -40,13 +40,13 @@
 
 - name: add system nameservers to docker options
   set_fact:
-    docker_dns_servers: "{{ docker_dns_servers + [item] }}"
-  with_items: "{{ system_nameservers.stdout_lines|default([]) }}"
+    docker_dns_servers: "{{ docker_dns_servers | system_nameservers.stdout_lines | unique }}"
```

Comment:
We don't even add upstream_dns_servers or nameservers vars here. The one and only item of concern is skydns_server/dns_server. 

union+unique drops items from the right hand side. If I put 10.233.0.2 in /etc/resolv.conf, it gets dropped because it's already in the beginning of the list from docker_dns_servers.